File: practica 1/sesion 3/DS_S3_C++Filters/Cliente.cpp
#include <exception>
#include <iostream>
using namespace std;

#include "Cliente.h"
#include "GestorFiltros.h"

void Cliente::pagar() {
	GestorFiltros gestor;
	gestor.filtrar(this->trabajo);
}


void Cliente::darParteTecnico(const int & amable,
	 	const int & profesional, const int & eficaz){

		trabajo->darParteTecnico(amable, profesional, eficaz);
}

Cliente::Cliente(string nombre_cliente) {
	this->nombre = nombre_cliente;
}

void Cliente::defineTrabajo(string & descripcion) {
	trabajo = new Trabajo(descripcion, this);
}

string Cliente::getNombre() {
	return this->nombre;
}

Trabajo * Cliente::getTrabajo() { return trabajo; }

File: practica 1/sesion 3/DS_S3_DartFilters/Cliente.dart
import 'Trabajo.dart';
import 'Tecnico.dart';
import 'GestorFiltros.dart';

class Cliente{
  String nombre;
  int edad;
  Trabajo trabajo;

  Cliente(String nombre){
    this.nombre = nombre;
  }

  void defineTrabajo(String desc){
    trabajo = new Trabajo(desc, this);
  }

  String getNombre(){
    return nombre;
  }

  Trabajo getTrabajo(){
    return trabajo;
  }

  void pagar(){
    GestorFiltros gestor = new GestorFiltros();
    gestor.filtrar(trabajo);
  }

  void darParteTecnico(amable, profesional, eficaz){
    trabajo.darParteTecnico(amable, profesional, eficaz);
  }
}